aten/src/ATen/TensorIterator.cpp

```diff
@@ -376,10 +376,8 @@ void TensorIteratorBase::compute_types(const TensorIteratorConfig& config) {
     // Checks all tensors are on the same device, if requested
     if (config.check_all_same_device_) {
       // Handles CPU scalars on CUDA kernels that support them
-      if (common_device.is_cuda() &&
-          config.allow_cpu_scalars_ &&
-          !op.is_output &&
-          op.tensor.dim() == 0 &&
+      if ((common_device.is_cuda() || common_device.is_xpu()) &&
```

Comment:
This change looks a little questionable; are you using TensorIterator to implement some XPU kernels?

Reply:
Yes. We are using TensorIterator in some XPU kernels.

Reply:
OK. FYI there's likely going to be some big refactors to TensorIterator (changing UI) coming this half. cc @heitorschueroff

Reply:
Thanks for the info.

@chengjunlu @arthuryuan1987